src/VirtoCommerce.SitemapsModule.Web/Controllers/Api/SitemapsModuleApiController.cs

```diff
@@ -329,21 +326,83 @@ public async Task<ActionResult<SitemapDownloadNotification>> DownloadSitemap(str
         [ApiExplorerSettings(IgnoreApi = true)]
         public Task BackgroundDownload(string storeId, string baseUrl, SitemapDownloadNotification notification)
         {
-            // We cannot use storeId.IndexOfAny(Path.GetInvalidFileNameChars()) != -1 to validate path because default
-            // sanitizer for Sonar Cube do not trust it, so we use Regex here with same logic. Check this out
-            // https://community.sonarsource.com/t/help-sonarcloud-with-understanding-the-usage-of-untrusted-and-tainted-input/9873/7
-            // Btw, we cannot move this to extansion or any method from here because sonar ignore any outer checks :(
-            if (!Regex.IsMatch(storeId, "^[a-zA-Z0-9-]+$"))
+            ArgumentNullException.ThrowIfNullOrWhiteSpace(storeId);
+
+            if (!Uri.TryCreate(baseUrl, UriKind.Absolute, out var correctUri))
             {
-                throw new ArgumentException($"Incorrect name of store {storeId}");
+                throw new ArgumentException($"Incorrect base URL {baseUrl}");
             }
 
+            return InnerBackgroundDownload(storeId, baseUrl, notification);
+        }
+
+        [HttpGet]
+        [Route("exportToStoreAssets")]
+        [Authorize(ModuleConstants.Security.Permissions.ExportToStoreAssets)]
+        public async Task<ActionResult<SitemapDownloadNotification>> ExportToStoreAssets(string storeId, string baseUrl)
+        {
+            var notification = new SitemapExportToAssetNotification(_userNameResolver.GetCurrentUserName())
+            {
+                Title = "Exporting sitemaps to store assets",
+                Description = "Processing exporting sitemaps to store assets..."
+            };
+
+            await _notifier.SendAsync(notification);
+
+            BackgroundJob.Enqueue(() => BackgroundExportToAssets(storeId, baseUrl, notification));
+
+            return Ok(notification);
+        }
+
+        [ApiExplorerSettings(IgnoreApi = true)]
+        public Task BackgroundExportToAssets(string storeId, string baseUrl, SitemapExportToAssetNotification notification)
```

Comment:
Do we really need this public method that does nothing but call a private method (except for checking the storeId)? Does it really have to be public? Maybe we can get rid of it and use only the private method? We also may convert the private method to public if necessary.

Reply:
Fixed. Move all job login to job service.